Win32-network/src/System/Win32/NamedPipes.hsc

```diff
@@ -147,3 +157,37 @@ foreign import ccall unsafe "windows.h CreateNamedPipeW"
                     -> DWORD
                     -> LPSECURITY_ATTRIBUTES
                     -> IO HANDLE
+
+
+-- | Timeout in milliseconds.
+--
+-- * 'nMPWAIT_USE_DEFAULT_WAIT' indicates that the timeout value passed to
+--   'createNamedPipe' should be used.
+-- * 'nMPWAIT_WAIT_FOREVER' - 'waitNamedPipe' will block forever, until a named
+--   pipe instance is available.
+--
+type TimeOut = DWORD
+#{enum TimeOut,
+ , nMPWAIT_USE_DEFAULT_WAIT = NMPWAIT_USE_DEFAULT_WAIT
+ , nMPWAIT_WAIT_FOREVER     = NMPWAIT_WAIT_FOREVER
+ }
+
+
+-- | Wait until a named pipe instance is available.  If there is no instance at
+-- hand before the timeout, it will error with 'ERROR_SEM_TIMEOUT', i.e.
+-- @invalid argument (The semaphore timeout period has expired)@
+--
+waitNamedPipe :: String  -- ^ pipe name
+              -> TimeOut -- ^ nTimeOut
+              -> IO ()
+waitNamedPipe name timeout =
+    withCString name $ \ c_name ->
+      failIfFalse_  "WaitNamedPipe" $
```

Comment:
Do we want it to fail on timeout?

We probably want 3 cases:
 * C `WaitNamedPipe` succeeds: Hs `waitNamedPipe` returns `True`
 * C `WaitNamedPipe` succeeds, errcode is `ERROR_SEM_TIMEOUT`: Hs `waitNamedPipe` returns `False`
 * C `WaitNamedPipe` succeeds, errcode is other: throw exception in the usual way.

Reply:
yes, just a correction when `waitNamedPipe` timeouts: it returns `0` and the error is `ERROR_SEM_TIMEOUT`.